src/utils/dataclass.py

```diff
@@ -21,26 +21,45 @@ def __post_init__(self):
             if isinstance(field.type, GenericAlias):
                 field_type = field.type.__args__[0]
                 if is_dataclass(field_type):
+                    factory = self.__get_dataclass_factory(field_type)
                     setattr(self, field.name,
                             field.type.__origin__(map(
-                                lambda x: field_type(**x) if not is_dataclass(x) else x,
+                                lambda x: factory(**x) if not is_dataclass(x) else x,
                                 getattr(self, field.name))))
             elif is_dataclass(field.type) and not is_dataclass(getattr(self, field.name)):
-                setattr(self, field.name, field.type(**getattr(self, field.name)))
+                factory = self.__get_dataclass_factory(field.type)
+                setattr(self, field.name, factory(**getattr(self, field.name)))
 
+    @staticmethod
+    def __get_dataclass_factory(field_type):
+        if issubclass(field_type, FromResponse):
+            return field_type.from_response
+        return field_type
 
-def list_of_dataclasses(_dataclass):
+@dataclass
+class FromResponse:
+    """
+    Class for extending dataclass with custom from_response method, ignored extra fields
+    """
+
+    @classmethod
+    def from_response(cls, **kwargs):
+        class_field_names = [field.name for field in fields(cls)]
+        return cls(**{k: v for k, v in kwargs.items() if k in class_field_names})
```

Comment:
This won't work if key will be integer. Maybe parse to str here or ignore int values

Reply:
JSON responses cannot contain integer keys. Therefore I assume that we don't receive integer keys here because casting json to dict already parses int keys to str